receiver/k8sobjectsreceiver/receiver.go

```diff
@@ -49,22 +55,69 @@ func newReceiver(params receiver.CreateSettings, config *Config, consumer consum
 		return nil, err
 	}
 
-	return &k8sobjectsreceiver{
-		client:   client,
-		setting:  params,
-		consumer: consumer,
-		objects:  config.Objects,
-		obsrecv:  obsrecv,
-		mu:       sync.Mutex{},
-	}, nil
+	objReceiver := &k8sobjectsreceiver{
+		client:         client,
+		setting:        params,
+		logger:         params.Logger,
+		consumer:       consumer,
+		objects:        config.Objects,
+		obsrecv:        obsrecv,
+		mu:             sync.Mutex{},
+		leaderElection: config.LeaderElection,
+	}
+
+	if config.LeaderElection.Enabled {
+		objReceiver.leaderElectionClient, err = config.getClient()
+		if err != nil {
+			return nil, err
+		}
+
+		// use component "type-name" if resource lock name not set.
+		if objReceiver.leaderElection.LockName == "" {
+			objReceiver.leaderElection.LockName = getLeaderElectionLockName(params.ID)
+		}
+	}
+
+	return objReceiver, nil
+}
+
+func (kr *k8sobjectsreceiver) startFunc(ctx context.Context) {
+	for _, object := range kr.objects {
+		kr.start(ctx, object)
+	}
+}
+
+func (kr *k8sobjectsreceiver) startInLeaderElectionMode(ctx context.Context) {
+	leaderLost := make(chan struct{})
+
+	kr.mu.Lock()
+	kr.stopperChanList = append(kr.stopperChanList, leaderLost)
+	kr.mu.Unlock()
+
+	lr, err := k8sconfig.NewLeaderElector(kr.leaderElection, kr.leaderElectionClient, func(_ context.Context) {
+		kr.logger.Info("this instance of the component was selected as the current leader")
```

Comment:
If there are two instances of the same component, each with their own lock names, would it be clear which one is which? Perhaps add the lock name to this log entry?

Reply:
luck name could not identify which the current instance is, how about pod ip or hostname ?